src/client/redux/clientSideGameExtras/clientSideGameExtras.ts

```diff
@@ -1,17 +1,31 @@
 import { createSlice, PayloadAction, Reducer } from '@reduxjs/toolkit';
+import { Card } from '@/types/cards';
 
 /**
  * Client-side held only game information, to store multiple things like:
  * - the attacking unit
  */
 interface ClientSideGameExtras {
     attackingUnit?: string; // cardId
+    lastPlayedCards?: Card[];
 }
 
+const NUM_LAST_PLAYED_CARDS = 20;
+
 export const clientSideGameExtrasSlice = createSlice({
     name: 'clientSideExtras',
-    initialState: { attackingUnit: undefined },
+    initialState: { attackingUnit: undefined, lastPlayedCards: [] },
     reducers: {
+        receiveLastPlayedCard(state, action: PayloadAction<Card>) {
+            state.lastPlayedCards = [
+                ...state.lastPlayedCards,
+                action.payload,
+            ].splice(-NUM_LAST_PLAYED_CARDS);
```

Comment:
whenever we use a number, it's best to make it explicit.  rather than .splice(-20)

Also right now, this feature is future-proofed to let it display up to 20 cards, so that we have the option to build a carousel of the last played cards (e.g. users complain about only being able to see 1 card back in history)